ruby/tests/api-test.cc

```diff
@@ -0,0 +1,172 @@
+
+using namespace std;
+using namespace leatherman::ruby;
+
+TEST_CASE("api::eval", "[ruby-api]") {
+    SECTION("can load api and evaluate ruby code") {
+        auto ruby = api::instance();
+        REQUIRE(ruby);
+        ruby->initialize();
+        REQUIRE(ruby->initialized());
+
+        REQUIRE(ruby->get_load_path().size() > 0);
+
+        REQUIRE(ruby->to_string(ruby->eval("'foo'")) == "foo");
+    }
+}
+
+TEST_CASE("api::is_*", "[ruby-api]") {
+    auto ruby = api::instance();
+    REQUIRE(ruby);
+    ruby->initialize();
+    REQUIRE(ruby->initialized());
+
+    SECTION("can correctly identify nil values") {
+        REQUIRE(ruby->is_nil(ruby->nil_value()));
+        REQUIRE_FALSE(ruby->is_nil(ruby->true_value()));
+    }
+
+    SECTION("can correctly identify true and false values") {
+        REQUIRE(ruby->is_true(ruby->true_value()));
+        REQUIRE_FALSE(ruby->is_true(ruby->false_value()));
+        REQUIRE(ruby->is_false(ruby->false_value()));
+        REQUIRE_FALSE(ruby->is_false(ruby->true_value()));
+    }
+
+    SECTION("can correctly identify strings") {
+        REQUIRE(ruby->is_string(ruby->utf8_value("'I'm a string'")));
+        REQUIRE_FALSE(ruby->is_string(1));
+    }
+
+    SECTION("can correctly identify symbols") {
+        REQUIRE(ruby->is_symbol(ruby->to_symbol("mysymbol")));
+        REQUIRE_FALSE(ruby->is_symbol(ruby->false_value()));
+    }
+
+    SECTION("can correctly identify numbers") {
+        REQUIRE(ruby->is_float(ruby->eval("1.5")));
+        REQUIRE_FALSE(ruby->is_float(ruby->utf8_value("foo")));
+
+        REQUIRE(ruby->is_fixednum(ruby->eval("2")));
+        REQUIRE_FALSE(ruby->is_fixednum(ruby->eval("1.5")));
+    }
+
+    SECTION("can correctly identify hashes") {
+        REQUIRE(ruby->is_hash(ruby->eval("{ 'red' => 2 }")));
+        REQUIRE_FALSE(ruby->is_hash(ruby->utf8_value("foo")));
+    }
+
+    SECTION("can correctly identify type") {
+        REQUIRE(ruby->is_a(ruby->eval("1"), ruby->eval("Integer")));
+        REQUIRE_FALSE(ruby->is_a(ruby->eval("'1'"), ruby->eval("Integer")));
+    }
+
+    SECTION("can correctly identify arrays") {
+        REQUIRE(ruby->is_array(ruby->eval("[1, 2, 3]")));
+        REQUIRE_FALSE(ruby->is_array(ruby->false_value()));
+    }
+}
+
+TEST_CASE("api::equals", "[ruby-api]") {
+    auto ruby = api::instance();
+    REQUIRE(ruby);
+    ruby->initialize();
+    REQUIRE(ruby->initialized());
+    SECTION("can correctly test boolean values for equality") {
+        REQUIRE(ruby->equals(ruby->true_value(), ruby->true_value()));
+        REQUIRE_FALSE(ruby->equals(ruby->true_value(), ruby->false_value()));
+    }
+
+    SECTION("can correctly test strings for equality") {
+        REQUIRE(ruby->equals(ruby->utf8_value("foo"), ruby->utf8_value("foo")));
+        REQUIRE_FALSE(ruby->equals(ruby->utf8_value("foo"), ruby->utf8_value("bar")));
+    }
+
+    SECTION("can correctly test numbers for equality") {
+        REQUIRE(ruby->equals(1, 1));
```

Comment:
No, I don't think so. Sorry about this, I realized later on that the numbers weren't doing what I thought they were. Guess I forgot to go back and fix them...